packages/web-components/fast-element/src/context.ts

```diff
@@ -0,0 +1,272 @@
+import type { Constructable } from "./interfaces.js";
+import { Metadata } from "./metadata.js";
+
+/**
+ * A Context object defines an optional initial value for a Context, as well as a name identifier for debugging purposes.
+ * @public
+ */
+export type Context<T> = {
+    readonly name: string;
+    readonly initialValue?: T;
+};
+
+/**
+ * A constant key that can be used to represent a Context dependency.
+ * The key can be used for context or DI but also doubles as a decorator for
+ * resolving the associated dependency.
+ * @beta
+ */
+export type ContextDecorator<T = any> = Readonly<Context<T>> &
+    PropertyDecorator &
+    ParameterDecorator;
+
+/**
+ * A Context object defines an optional initial value for a Context, as well as a name identifier for debugging purposes.
+ * The FASTContext can also be used as a decorator to declare context dependencies or as a key for DI.
+ * @beta
+ */
+export type FASTContext<T> = ContextDecorator<T> & {
+    get(target: EventTarget): T;
+    request(target: EventTarget, callback: ContextCallback<T>, multiple?: boolean): void;
+    handle(target: EventTarget, callback: (event: ContextEvent<FASTContext<T>>) => void);
+};
+
+/**
+ * A strategy that controls how all Context.request API calls are handled.
+ * @remarks
+ * By default this is handled via Context.dispatch, which dispatched a ContextEvent.
+ * @beta
+ */
+export type FASTContextRequestStrategy = <T extends UnknownContext>(
+    target: EventTarget,
+    context: T,
+    callback: ContextCallback<ContextType<T>>,
+    multiple
+) => void;
+
+const contextEventType = "context-request";
+let requestStrategy: FASTContextRequestStrategy;
+
+/**
+ * Enables using the W3C Community Context protocol.
```

Comment:
```suggestion
 * Enables using the {@link https://github.com/webcomponents-cg/community-protocols/blob/main/proposals/context.md | W3C Community Context protocol.}
```

Reply:
I thought it might be worthwhile to link the *source* of that protocol so folks can learn more.